packages/server/src/api/muster/muster.compliance.ts

```diff
@@ -45,29 +46,71 @@ export function toMusterIntermediateCompliance(rosters: Roster[]): IntermediateM
  * TBD later....
  * @param observations
  * @param musteringOpportunities
- * @param musterComplianceReport
+ * @param rosterForMusterCompliance
  */
-export function calculateMusterCompliance(
-  observations: RecordedObservations[],
-  musteringOpportunities: MusteringOpportunities,
-  musterComplianceReport: IntermediateMusterCompliance[],
-): MusterCompliance[] {
-  return [];
+export function calculateMusterCompliance(observations: RecordedObservations[], musteringOpportunities: MusteringOpportunities, rosterForMusterCompliance: RosterForMusterCompliance[]): MusterCompliance[] {
+  return rosterForMusterCompliance.map(roster => {
+    const edipi = roster.edipi;
+    const unitId = roster.unitId;
+    const unitMusteringOpportunities = musteringOpportunities[unitId];
+
+    if (!unitMusteringOpportunities || unitMusteringOpportunities.length === 0) {
+      return { totalMusters: 0, mustersReported: 0, musterPercent: 100, ...roster };
+    }
+
+    const totalMusters = unitMusteringOpportunities.length;
+
+    const edipiObservations = filterObservationsByEdipi(observations, edipi);
+    const mustersReported = calculateIndividualMusterCompliance(edipiObservations, unitMusteringOpportunities);
+    const musterPercent = (100 * mustersReported) / totalMusters;
```

Comment:
order of ops doesn't matter with multiplication and division ;), generally like to see the mult come second when normalizing values but may just be me lol

Reply:
Correct, but I like the explicit style. ;-)

Reply:
The order in this case doesn't matter but I like to keep the division last as it may result in some cases a grater precision of the result but it all depends on the numbers used.